File: examples/companion-packages/php-composer-app/src/MatchPreferences.php
<?php

declare(strict_types=1);

namespace AgentXM\Examples\PawMatch;

final class MatchPreferences
{
    public function __construct(
        public readonly bool $hasKids = false,
        public readonly bool $quietHome = false,
        public readonly bool $active = false,
        public readonly bool $firstTime = false,
        public readonly bool $multiplePets = false,
        public readonly bool $smallHome = false,
    ) {
    }

    /**
     * @return list<string>
     */
    public function activeFlags(): array
    {
        $flags = [];
        if ($this->hasKids) {
            $flags[] = 'has-kids';
        }
        if ($this->quietHome) {
            $flags[] = 'quiet-home';
        }
        if ($this->active) {
            $flags[] = 'active';
        }
        if ($this->firstTime) {
            $flags[] = 'first-time';
        }
        if ($this->multiplePets) {
            $flags[] = 'multiple-pets';
        }
        if ($this->smallHome) {
            $flags[] = 'small-home';
        }

        return $flags;
    }

    public function isEmpty(): bool
    {
        return $this->activeFlags() === [];
    }
}

File: examples/companion-packages/maven-java-app/src/main/java/ai/agentxm/examples/pawmatch/MatchPreferences.java
package ai.agentxm.examples.pawmatch;

import java.util.LinkedHashSet;
import java.util.Set;

/** User-supplied preferences for the {@code match} command. */
public record MatchPreferences(
        boolean hasKids,
        boolean quietHome,
        boolean active,
        boolean firstTime,
        boolean multiplePets,
        boolean smallHome) {

    public static final MatchPreferences EMPTY =
            new MatchPreferences(false, false, false, false, false, false);

    public Set<String> activeFlags() {
        Set<String> flags = new LinkedHashSet<>();
        if (hasKids) flags.add("has-kids");
        if (quietHome) flags.add("quiet-home");
        if (active) flags.add("active");
        if (firstTime) flags.add("first-time");
        if (multiplePets) flags.add("multiple-pets");
        if (smallHome) flags.add("small-home");
        return flags;
    }

    public boolean isEmpty() {
        return !hasKids && !quietHome && !active && !firstTime && !multiplePets && !smallHome;
    }
}
